internal/broker/broker.go

```diff
@@ -366,24 +364,27 @@ func (b *Broker) IsAuthenticated(sessionID, authenticationData string) (string,
 	defer b.CancelIsAuthenticated(sessionID)
 
 	authDone := make(chan struct{})
-	var access, data string
+	var result isAuthenticatedResult
 	go func() {
-		access, data = b.handleIsAuthenticated(ctx, &session, authData)
+		result = b.handleIsAuthenticated(ctx, &session, authData)
 		close(authDone)
 	}()
 
 	select {
 	case <-authDone:
 	case <-ctx.Done():
-		return AuthCancelled, `{"message": "authentication request cancelled"}`, ctx.Err()
+		result := isAuthenticatedResult{Message: "authentication request cancelled"}
+		data, _ := result.toJSON()
```

Comment:
I know the message is just above, but it's a good practice to add a comment explaining why we are ignoring the error.